tsl/test/expected/chunk_api-12.out

```diff
@@ -517,3 +567,211 @@ WARNING:  insufficient number of data nodes for distributed hypertable "disttabl
+-------
+     1
+(1 row)
+
+DROP TABLE chunkapi;
+CREATE TABLE chunkapi (time timestamptz not null, device int, temp float);
+SELECT * FROM create_hypertable('chunkapi', 'time', 'device', 3);
+ hypertable_id | schema_name | table_name | created 
+---------------+-------------+------------+---------
+             5 | public      | chunkapi   | t
+(1 row)
+
+SELECT count(*) FROM 
+   _timescaledb_internal.create_chunk_table('chunkapi', :'SLICES', :'CHUNK_SCHEMA', :'CHUNK_NAME');
+ count 
+-------
+     1
+(1 row)
+
+-- Test create_chunk_table if a colliding chunk exists
+DROP TABLE chunkapi;
+CREATE TABLE chunkapi (time timestamptz not null, device int, temp float);
+SELECT * FROM create_hypertable('chunkapi', 'time', 'device', 3);
+ hypertable_id | schema_name | table_name | created 
+---------------+-------------+------------+---------
+             6 | public      | chunkapi   | t
+(1 row)
+
+INSERT INTO chunkapi VALUES ('2018-01-01 05:00:00-8', 1, 23.4);
+\set ON_ERROR_STOP 0
+SELECT _timescaledb_internal.create_chunk_table('chunkapi', :'SLICES', :'CHUNK_SCHEMA', :'CHUNK_NAME');
+ERROR:  chunk table creation failed due to dimension slice collision
+\set ON_ERROR_STOP 1
+DROP TABLE chunkapi;
+-- Test create_chunk_table with tablespaces
+\c :TEST_DBNAME :ROLE_SUPERUSER
+SET client_min_messages = ERROR;
+DROP TABLESPACE IF EXISTS tablespace1;
+DROP TABLESPACE IF EXISTS tablespace2;
+SET client_min_messages = NOTICE;
+CREATE TABLESPACE tablespace1 OWNER :ROLE_DEFAULT_PERM_USER LOCATION :TEST_TABLESPACE1_PATH;
+CREATE TABLESPACE tablespace2 OWNER :ROLE_DEFAULT_PERM_USER LOCATION :TEST_TABLESPACE2_PATH;
+\c :TEST_DBNAME :ROLE_DEFAULT_PERM_USER
+-- Use the space partition to calculate the tablespace id to use
+CREATE TABLE chunkapi (time timestamptz not null, device int, temp float);
+SELECT * FROM create_hypertable('chunkapi', 'time', 'device', 3);
+ hypertable_id | schema_name | table_name | created 
+---------------+-------------+------------+---------
+             7 | public      | chunkapi   | t
+(1 row)
+
+SELECT attach_tablespace('tablespace1', 'chunkapi');
+ attach_tablespace 
+-------------------
+ 
+(1 row)
+
+SELECT attach_tablespace('tablespace2', 'chunkapi');
+ attach_tablespace 
+-------------------
+ 
+(1 row)
+
+SELECT count(*) FROM 
+   _timescaledb_internal.create_chunk_table('chunkapi', :'SLICES', :'CHUNK_SCHEMA', :'CHUNK_NAME');
+ count 
+-------
+     1
+(1 row)
+
+SELECT tablespace FROM pg_tables WHERE tablename = :'CHUNK_NAME';
+ tablespace  
+-------------
+ tablespace1
+(1 row)
+
+DROP TABLE chunkapi;
+-- Use the time partition to calculate the tablespace id to use
+CREATE TABLE chunkapi (time timestamptz not null, device int, temp float);
+SELECT * FROM create_hypertable('chunkapi', 'time');
+ hypertable_id | schema_name | table_name | created 
+---------------+-------------+------------+---------
+             8 | public      | chunkapi   | t
+(1 row)
+
+INSERT INTO chunkapi VALUES ('2018-01-01 05:00:00-8', 1, 23.4);
```

Comment:
I would also like to see a test where the chunk does not collide due to being a different space partition (i.e., it still overlaps in the time dimension). For instance, if you instead insert this row:
```
INSERT INTO chunkapi VALUES ('2018-01-01 05:00:00-8', 2, 23.4);
```

Then, assuming "device 2" is another space partition than "device 1", the created chunk table should not collide.